Plutarch/Api/V1.hs

```diff
@@ -533,6 +602,8 @@ instance PIsDataRepr PDCert where
 newtype PMap (a :: PType) (b :: PType) (s :: S)
   = PMap (Term s (PBuiltinMap a b))
 
+-- TODO: PMap needs instances for PConstant/PLift, PlutusType, and PIsData.
```

Comment:
Currently getting a bunch of type checking errors when I try doing this. Want to tackle it on a separate PR if possible.

Reply:
I was running into many headaches with this too...
Manual instances would be IMO acceptable if necessary.